lib/rules/base.js

```diff
@@ -80,6 +91,46 @@ module.exports = class BaseRule {
     return config;
   }
 
+  /**
+   * Sets the severity of the rule based on user configuration
+   * and returns the config based on the severity OR as defined by the user.
+   * Eg:
+   * {'no-implicit-this': 'warn'}
+   * This would set the severity to 1 and return `true`.
+   * { 'no-implicit-this': { "severity": "warn", "config": "allow": [ "fooData" ] } }
+   * This would set the severity of the rule to 1 and return the user defined config back.
+   * { 'no-implicit-this': { "severity": "warn"} }
+   * This would set the severity of the rule to 1 and return true.
+   * @param {*} configData
+   */
+  processConfigForSeverity(configData) {
```

Comment:
I don't think we need this method to be on the rule itself, I'd like it to be part of `get-config` instead.

Basically, I'd like `get-config.js` utility to return `config.rules` that is in the internal structure that you mentioned in the initial description:

```
rules: {
  'rule-name': {
    severity: 'off' | 'warn' | 'error',
    config: someUserlandValue
  }
}
```

Then when the rule is instantiated (in `lib/linter.js`'s `buildRule`) we pass options that include `severity` and `config` (aka, the same thing we pass today).

This will require changing a few tests that are asserting on the results of `get-config`, but shouldn't be _too_ bad...